lib/gui.py

```diff
@@ -279,6 +275,48 @@ def start_wordlist_gen(url: str, file: str, min: int, max: int):
     copy_button.pack()
 
 
+def make_hash(main_window):
+    hash_window = tk.Toplevel(main_window, bg=DEFAULT_BG_COLOR, width=DEFAULT_WIDTH, height=DEFAULT_HEIGHT)
+    hash_window.resizable(False, False)
+    hash_window.title("File and text hasher")
+
+    hash_frame = tk.Frame(hash_window)
+    hash_frame.pack(fill=tk.BOTH, expand=True)
+
+    input_to_hash = tk.StringVar(value=None)
+    algorithm = tk.StringVar(value="SHA256")
+    buf_size = tk.StringVar(value=65536)
+    output = tk.StringVar(value="")
+
+    def forward_and_insert_hash_result(input, buf_size, algorithm, output):
+        result = hash_input(input=input, buf_size=buf_size, algorithm=algorithm, output=output)
+        result_label.config(text=result)
+        copy_button.config(command=lambda: copy_to_clipboard(result))
+        copy_button.pack()
+
+
+    entries = [
+        tk.Label(hash_frame, text="File/String to hash"),
+        EntryWithPlaceholder(hash_frame, "", textvariable=input_to_hash),
+        tk.Label(hash_frame, text="Choose hashing algorithm:"),
+        EntryWithPlaceholder(hash_frame, "", textvariable=algorithm),
+        tk.Label(hash_frame, text="Prefered buf size"),
+        EntryWithPlaceholder(hash_frame, "", textvariable=buf_size),
+        tk.Label(hash_frame, text="Output file name. If submitted empty, program will print your hash below with copy button"),
+        EntryWithPlaceholder(hash_frame, "", textvariable=output),
+        tk.Button(
+            hash_frame,
+            text="Submit Input",
+            command=lambda: forward_and_insert_hash_result(input=input_to_hash.get(), buf_size=buf_size.get(), algorithm=algorithm.get(), output=output.get()),
+        ),
+    ]
+    for entry in entries:
+        entry.pack(fill=tk.X, expand=True)
+
+    result_label = tk.Label(hash_frame, text="")
```

Comment:
I think using a `Text` instead of `Label` is better. What do you think

Reply:
why it's better?

Reply:
Because Label was designed to describe what the widget below is, not to put data in. The Text widget also allows to select text with mouse and copy it that way